Runner/utils/lib_common.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+# Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+# POSIX helpers that defer to functestlib.sh where available.
+
+# --- fallbacks only if functestlib wasn't sourced ---
+if ! command -v log_info >/dev/null 2>&1; then
+  log_info() { printf '[INFO] %s\n' "$*"; }
+fi
+if ! command -v log_warn >/dev/null 2>&1; then
+  log_warn() { printf '[WARN] %s\n' "$*" >&2; }
+fi
+if ! command -v log_error >/dev/null 2>&1; then
+  log_error() { printf '[ERROR] %s\n' "$*" >&2; }
+fi
+if ! command -v log_fail >/dev/null 2>&1; then
+  log_fail() { printf '[FAIL] %s\n' "$*" >&2; }
+fi
+if ! command -v log_skip >/dev/null 2>&1; then
+  log_skip() { printf '[SKIP] %s\n' "$*"; }
+fi
+
+die() { log_error "$*"; exit 1; }
+
+# --- cmd exists ---
+have_cmd() { command -v "$1" >/dev/null 2>&1; }
+
+# --- root / sudo (kept tiny; functestlib may not provide them) ---
+SUDO=""
+is_root() { [ "$(id -u 2>/dev/null || echo 1)" -eq 0 ]; }
+need_root() {
+  if ! is_root; then
+    if have_cmd sudo; then SUDO="sudo -E"; else die "Root required and 'sudo' not found."; fi
+  fi
+}
+
+# --- ensure dir ---
+ensure_dir() {
+  d="$1"
+  if [ ! -d "$d" ]; then
+    mkdir -p "$d" 2>/dev/null || { need_root; $SUDO mkdir -p "$d" || return 1; }
+  fi
+}
+
+# --- OS detect (Debian/Ubuntu) ---
+detect_os_like() {
+  if [ -r /etc/os-release ]; then
+    # shellcheck source=/dev/null
```

Comment:
meh :)

Would this make shellcheck happy enough?
`(. /etc/os-release && echo "${ID_LIKE:-$ID}")`